api/fees/data.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package fees
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/vechain/thor/v2/block"
+	"github.com/vechain/thor/v2/cache"
+	"github.com/vechain/thor/v2/chain"
+	"github.com/vechain/thor/v2/thor"
+)
+
+type FeeCacheEntry struct {
+	baseFee       *hexutil.Big
+	gasUsedRatio  float64
+	parentBlockID thor.Bytes32
+}
+type FeesData struct {
+	repo  *chain.Repository
+	cache *cache.PrioCache
+}
+
+func newFeesData(repo *chain.Repository, fixedSize uint32) *FeesData {
+	return &FeesData{
+		repo:  repo,
+		cache: cache.NewPrioCache(int(fixedSize)),
+	}
+}
+
+func getBaseFee(baseFee *big.Int) *hexutil.Big {
+	if baseFee != nil {
+		return (*hexutil.Big)(baseFee)
+	}
+	return (*hexutil.Big)(big.NewInt(0))
+}
+
+func (fd *FeesData) pushToCache(header *block.Header) {
+	feeCacheEntry := &FeeCacheEntry{
+		baseFee:       getBaseFee(header.BaseFee()),
+		gasUsedRatio:  float64(header.GasUsed()) / float64(header.GasLimit()),
+		parentBlockID: header.ParentID(),
+	}
+	fd.cache.Set(header.ID(), feeCacheEntry, float64(header.Number()))
+}
+
+// resolveRange resolves the base fees and gas used ratios for the given block range.
+// Assumes that the boundaries (newest block - block count) are correct and validated beforehand.
+func (fd *FeesData) resolveRange(newestBlockSummary *chain.BlockSummary, blockCount uint32) (thor.Bytes32, []*hexutil.Big, []float64, error) {
+	newestBlockID := newestBlockSummary.Header.ID()
+
+	baseFees := make([]*hexutil.Big, blockCount)
+	gasUsedRatios := make([]float64, blockCount)
+
+	var oldestBlockID thor.Bytes32
+	for i := blockCount; i > 0; i-- {
+		oldestBlockID = newestBlockID
+		fees, _, found := fd.cache.Get(newestBlockID)
```

Comment:
how about a function called `fd.getFeeEentry(blockID thor.bytes32) (*FeeCacheEntry, error)` to warp up a `get or load` logic?

Reply:
I could do it, but, it is not already pretty simple and straightforward this method? I think it could harm readability the fact that we need to check another method to see what is happening.

Reply:
The computation has been run twice in `pushToCache`  and outside, which I would suggest being avoided.

Reply:
With "computation" you mean the assignment for `baseFees`, `gasUsedRatios` and so on?

Reply:
I can make `pushToCache` return the `FeeCacheEntry` (and assign it to `fees`) so the below computation does not happen and it is just 

```
		baseFees[i-1] = getBaseFee((*big.Int)(fees.(*FeeCacheEntry).baseFee))
		gasUsedRatios[i-1] = fees.(*FeeCacheEntry).gasUsedRatio

		newestBlockID = fees.(*FeeCacheEntry).parentBlockID
```

let me know what you think

Reply:
Would you mind to put the logic of `pushToCache` into the code above?

Reply:
Done